<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Normalize and detect differences between Hi-C datasets when replicates 
  of each experimental condition are available • multiHiCcompare</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Normalize and detect differences between Hi-C datasets when replicates 
  of each experimental condition are available">
<meta property="og:description" content="multiHiCcompare provides functions for joint normalization 
    and difference detection in multiple Hi-C datasets. This extension of
    the original HiCcompare package now allows for Hi-C experiments with 
    more than 2 groups and multiple samples per group. multiHiCcompare 
    operates on processed Hi-C data in the form of sparse upper triangular 
    matrices. It accepts four column (chromosome, region1, region2, IF)
    tab-separated text files storing chromatin interaction matrices. 
    multiHiCcompare provides cyclic loess and fast loess (fastlo) methods
    adapted to jointly normalizing Hi-C data. Additionally, it provides 
    a general linear model (GLM) framework adapting the edgeR package to detect
    differences in Hi-C data in a distance dependent manner. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">multiHiCcompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/multiHiCcompare.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/juiceboxVisualization.html">Visualizing multiHiCcompare results in Juicebox</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dozmorovlab/multiHiCcompare/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="multihiccompare" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#multihiccompare" class="anchor"></a><a href="https://dozmorovlab.github.io/multiHiCcompare/">multiHiCcompare</a>
</h1></div>
<!-- badges: start -->

<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p><code>multiHiCcompare</code> is an extension of the original <a href="http://bioconductor.org/packages/HiCcompare/">HiCcompare R package</a>. <code>multiHiCcompare</code> provides functions for the joint normalization and comparison of complex Hi-C experiments. <code>multiHiCcompare</code> operates on processed Hi-C data in the form of sparse upper triangular matrices.</p>
<p><code>multiHiCcompare</code> accepts four-column text files storing chromatin interaction matrices in a sparse matrix format. There are many sources of public Hi-C data such as the <a href="http://aidenlab.org/data.html">Aiden Lab</a> (<code>.hic</code> files) and the <a href="http://cooler.readthedocs.io/en/latest/index.html">Mirnylab FTP site</a> (<code>.cool</code> files). <code>multiHiCcompare</code> performs differential chromatin interaction analysis between two biological conditions, one or multiple Hi-C matrices per condition.</p>
<p><code>multiHiCcompare</code> implements a cyclic loess joint normalization algorithm to remove bias between multiple Hi-C datasets and prepare them for comparison. <code>multiHiCcompare</code> also provides a general linear model-based difference detection method, implemented in the <code>edgeR</code> R package.</p>
<p>The main functions of <code>multiHiCcompare</code> are:</p>
<ul>
<li>
<code><a href="reference/cyclic_loess.html">cyclic_loess()</a></code> and <code><a href="reference/fastlo.html">fastlo()</a></code> for performing joint normalization.</li>
<li>
<code><a href="reference/hic_exactTest.html">hic_exactTest()</a></code> and <code><a href="reference/hic_glm.html">hic_glm()</a></code> for performing comparisons between experimental conditions.</li>
</ul>
<p>Some example Hi-C data are included in the package. Refer to the <code>multiHiCcompare</code> vignette for full usage instructions, <code><a href="articles/multiHiCcompare.html">vignette("multiHiCcompare")</a></code></p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>First, make sure you have all dependencies installed in R.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'dplyr'</span>, <span class="st">'data.table'</span>, <span class="st">'devtools'</span>, <span class="st">'qqman'</span>, <span class="st">'metap'</span>, <span class="st">'pheatmap'</span>, <span class="st">'pbapply'</span><span class="op">)</span><span class="op">)</span>

<span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"BiocParallel"</span>, <span class="st">"HiCcompare"</span>, <span class="st">"edgeR"</span>, <span class="st">"GenomicRanges"</span>, <span class="st">"GenomeInfoDbData"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>To install <code>multiHiCcompare</code> from Bioconductor, use the following commands.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># It is recommended to use the GitHub release until the next Bioconductor update</span>
<span class="co"># BiocManager::install("multiHiCcompare")</span>
<span class="co"># library(multiHiCcompare)</span></code></pre></div>
<p>Or to install the latest version of <code>multiHiCcompare</code> directly from the GitHub.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="co"># Stable release</span>
<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'dozmorovlab/multiHiCcompare'</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># Developmental version</span>
<span class="co"># install_github('jstansfield0/multiHiCcompare', build_vignettes = TRUE)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dozmorovlab/multiHiCcompare">multiHiCcompare</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>To use <code>multiHiCcompare</code>, you will first need to obtain some Hi-C data. Data is available from the sources listed in the overview, along with many others. You will need to extract the data and read it into R as either a 3 column sparse upper triangular matrix and then combine it with an additional column for the chromosome. Hi-C data ready to be used in <code>multiHiCcompare</code> should look like the following:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a>  chr  region1  region2 IF</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true"></a><span class="dv">1</span>  <span class="dv">22</span> <span class="dv">16000000</span> <span class="dv">16000000</span> <span class="dv">11</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true"></a><span class="dv">2</span>  <span class="dv">22</span> <span class="dv">16100000</span> <span class="dv">16100000</span>  <span class="dv">1</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true"></a><span class="dv">3</span>  <span class="dv">22</span> <span class="dv">16200000</span> <span class="dv">16200000</span>  <span class="dv">3</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true"></a><span class="dv">4</span>  <span class="dv">22</span> <span class="dv">16300000</span> <span class="dv">16300000</span> <span class="dv">15</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true"></a><span class="dv">5</span>  <span class="dv">22</span> <span class="dv">16400000</span> <span class="dv">16400000</span>  <span class="dv">3</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true"></a><span class="dv">6</span>  <span class="dv">22</span> <span class="dv">16400000</span> <span class="dv">16500000</span>  <span class="dv">1</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true"></a>...</span></code></pre></div>
<p>The four columns correspond to the chromosome, the start location of the first interacting region (in base pairs), the start location of the second interacting region, and the interaction frequency (IF) of the interaction.</p>
<p>Below is an example analysis of a Hi-C experiment.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load data</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dozmorovlab/multiHiCcompare">multiHiCcompare</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"r1"</span>, <span class="st">"r2"</span>, <span class="st">"r3"</span>, <span class="st">"r4"</span><span class="op">)</span>

<span class="co"># make hicexp object</span>
<span class="va">hicexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/make_hicexp.html">make_hicexp</a></span><span class="op">(</span><span class="va">r1</span>, <span class="va">r2</span>, <span class="va">r3</span>, <span class="va">r4</span>, groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>

<span class="co"># jointly normalize data</span>
<span class="va">hicexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cyclic_loess.html">cyclic_loess</a></span><span class="op">(</span><span class="va">hicexp</span><span class="op">)</span>

<span class="co"># compare groups</span>
<span class="va">hicexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/hic_exactTest.html">hic_exactTest</a></span><span class="op">(</span><span class="va">hicexp</span><span class="op">)</span>

<span class="co"># view manhattan plot of results</span>
<span class="fu"><a href="reference/manhattan_hicexp.html">manhattan_hicexp</a></span><span class="op">(</span><span class="va">hicexp</span><span class="op">)</span></code></pre></div>
</div>
<div id="citation" class="section level2">
<h2 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h2>
<p>Stansfield, John C, Kellen G Cresswell, and Mikhail G Dozmorov. <a href="https://doi.org/10.1093/bioinformatics/btz048">MultiHiCcompare: Joint Normalization and Comparative Analysis of Complex Hi-C Experiments</a>. <em>Bioinformatics</em>, January 22, 2019.</p>
</div>
<div id="learn-more" class="section level2">
<h2 class="hasAnchor">
<a href="#learn-more" class="anchor"></a>Learn more</h2>
<p>Stansfield, John C., Duc Tran, Tin Nguyen, and Mikhail G. Dozmorov. <a href="https://doi.org/10.1002/cpbi.76">R Tutorial: Detection of Differentially Interacting Chromatin Regions From Multiple Hi-C Datasets</a>. Current Protocols in Bioinformatics, May 2019</p>
<p><a href="https://github.com/mdozmorov/HiCcompareWorkshop">HiCcompareWorkshop</a> - “Detection of Differentially Interacting Chromatin Regions From Multiple Hi-C Datasets” workshop presented on Bioconductor 2020 conference</p>
</div>
<div id="contributions--support" class="section level2">
<h2 class="hasAnchor">
<a href="#contributions--support" class="anchor"></a>Contributions &amp; Support</h2>
<p><a href="https://github.com/dozmorovlab/HiCcompare/issues">issue</a> for any of these or contact the author directly: <a href="https://github.com/jstansfield0">@jstansfield0</a> (<a href="mailto:stansfieldjc@vcu.edu" class="email">stansfieldjc@vcu.edu</a>)</p>
</div>
<div id="contributors" class="section level2">
<h2 class="hasAnchor">
<a href="#contributors" class="anchor"></a>Contributors</h2>
<p>Authors: <a href="https://github.com/jstansfield0">@jstansfield0</a> (<a href="mailto:stansfieldjc@vcu.edu" class="email">stansfieldjc@vcu.edu</a>) &amp; <a href="https://github.com/mdozmorov">@mdozmorov</a> (<a href="mailto:mikhail.dozmorov@vcuhealth.org" class="email">mikhail.dozmorov@vcuhealth.org</a>)</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/multiHiCcompare">https://​www.bioconductor.org/​packages/​multiHiCcompare</a>
</li>
<li>Browse source code at <br><a href="https://github.com/dozmorovlab/multiHiCcompare/">https://​github.com/​dozmorovlab/​multiHiCcompare/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/dozmorovlab/multiHiCcompare/issues">https://​github.com/​dozmorovlab/​multiHiCcompare/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing multiHiCcompare</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>John Stansfield <br><small class="roles"> Maintainer </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://bioconductor.org/checkResults/release/bioc-LATEST/multiHiCcompare"><img src="http://www.bioconductor.org/shields/build/release/bioc/multiHiCcompare.svg" alt="BioC status"></a></li>
<li><a href="https://www.tidyverse.org/lifecycle/#stable"><img src="https://img.shields.io/badge/lifecycle-stable-brightgreen.svg" alt="Lifecycle: stable"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by John Stansfield.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
